src/wms/src/main/java/org/geoserver/wms/GetFeatureInfo.java

```diff
@@ -58,8 +60,14 @@ private FeatureCollectionType buildResults(List<FeatureCollection> results) {
 
     @SuppressWarnings("rawtypes")
     private List<FeatureCollection> execute(GetFeatureInfoRequest request) throws Exception {
+
         final List<MapLayerInfo> requestedLayers = request.getQueryLayers();
         FeatureInfoRequestParameters requestParams = new FeatureInfoRequestParameters(request);
+        // if url has a clip param and the clicked point is outside
+        // then abort and return empty list
+        if (!ClipWMSGetMapCallBack.validateClipParam(requestParams)) {
```

Comment:
I don't believe this is quite right. A GetFeatureInfo uses a search buffer and by default a full rendering to figure out what was _depicted_ at the point clicked. "Depicted" is key, could have been a large point symbolizer, or a small one with an offset, whose source point is outside of the clip. Conversely, a large point symbolizer could allow to render outside of the clip a symbol whose origin is inside the clip.
So this test is ineffective, the data should be clipped like in GetMap instead. Maybe you can roll a super-interface of GetMapCallback that allows to do this job, and the callback can implement it? GetFeatureInfo.java would have to call the callbacks of course.

Reply:
understood, I will clip the getFeatureInfo response just like Get Map.